server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -273,9 +275,15 @@ private static LeafSlice[] computeSlices(List<LeafReaderContext> leaves, int min
     @Override
     public <C extends Collector, T> T search(Query query, CollectorManager<C, T> collectorManager) throws IOException {
         final C firstCollector = collectorManager.newCollector();
-        // Take advantage of the few extra rewrite rules of ConstantScoreQuery when score are not needed.
-        query = firstCollector.scoreMode().needsScores() ? rewrite(query) : rewrite(new ConstantScoreQuery(query));
-        final Weight weight = createWeight(query, firstCollector.scoreMode(), 1);
+        final Weight weight;
+        try {
+            // Take advantage of the few extra rewrite rules of ConstantScoreQuery when score are not needed.
+            query = firstCollector.scoreMode().needsScores() ? rewrite(query) : rewrite(new ConstantScoreQuery(query));
```

Comment:
I am not familiar with the rewrite logic but it makes sense

Reply:
I had a look and the call to `SearchContext#rewrittenQuery` cannot timeout because we have not added the timeout to the IndexSearcher yet.

Regarding the noop, I think what is important here is the `ConstantScoreQuery` and not sure if that can have some effect on the rewrite.

Reply:
good point, I will dig further, but I think this is not something that require further work in this PR, at first glance. I will try to write unit tests the leverage the timeout in the different places where it can trigger a TimeExceededException.

Reply:
ok, I debated for a bit about whether we should remove the rewrite catch, because it is effectively not needed as we can't possibly time out there. It can be misleading.

As for createWeight, I did not find a way to test partial results in there, but setting the flat timedOut seems correct.

I will leave things as-is for now, we will have to get back to it and clean it up later.